src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,103 @@
+package duke;
+
+import java.util.ArrayList;
+
+/**
+ * Encapsulates a container for Tasks.
+ *
+ * @author Sean Chin Jun Kai.
+ */
+public class TaskList {
+    private ArrayList<Task> tasks;
+
+    /**
+     * Constructor for initialising new TaskList.
+     */
+    public TaskList() {
+        this.tasks = new ArrayList<>();
+    }
+
+    /**
+     * Adds task to the TaskList.
+     *
+     * @param task Task to be added.
+     */
+    public void addTask(Task task) {
+        this.tasks.add(task);
+    }
+
+    /**
+     * Retrieves a task from the TaskList based on given ID.
+     *
+     * @param taskNumber ID of the task in the TaskList.
+     * @return chosen Task.
+     * @throws DukeException if given ID does not exist in TaskList.
+     */
+    public Task getTask(int taskNumber) throws DukeException {
```

Comment:
good use of dukeException to convert the default array out of bounds exception to a custom one